### PR TITLE
Add Ledger Stax support

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

**Comment by xchapron-ledger** [comment]
@cryptoBeliever @gimre-xymcity @Jaguar0625 Here is what I did for the porting to the new Ledger hardware wallet: Ledger Stax.
For now the SDK is not public, hence the draft and the missing jobs in CI.
But still you can have a look at the code changes and the corresponding screenshots in the tests.
Please do no hesitate to propose changes if some are needed. Specifically regarding the icons. I made them myself by toggling pixels on Gimp and I'm pretty sure someone with more skills can do better :)

**Comment by xchapron-ledger** [comment]
@cryptoBeliever @gimre-xymcity @Jaguar0625 It's now fully ready! Please have a look :)

**Comment by xchapron-ledger** [comment]
Won't fix the clang warning. Was already there and should be address in a separate PR.

**Review by gimre-xymcity — APPROVED** [review]

File: src/ui/address/address_ui.c [hunk 1]
```diff
@@ -22,12 +22,20 @@
...
 #include "nem/nem_helpers.h"
 #include "glyphs.h"
+#ifdef HAVE_NBGL
+#include "nbgl_use_case.h"
+#endif
 
+#ifdef HAVE_BAGL
 extern char fieldValue[MAX_FIELD_LEN];
+#else
+static char fieldValue[MAX_FIELD_LEN];
```
Comment (sgliner-ledger) [inline src/ui/address/address_ui.c:32]: Why do you need to do this ? :thinking:

**Review by sgliner-ledger — APPROVED** [review]
Very good work ! :rocket:

File: src/ui/address/address_ui.c (hunk 1 above)
Comment (xchapron-ledger) [reply src/ui/address/address_ui.c:32]: On Nano device `fieldName` and `fieldValue` are declared in `review_menu.c` and also used in `address_ui.c` hence it is shared to spare RAM.
On Stax, I only need it for `address_ui.c` as I need something more complex for `review_menu.c`. Hence the declaration here.

**Pull request merged**